ubi_manifest/app/api.py

```diff
@@ -1,21 +1,120 @@
 import json
+from datetime import datetime, timedelta
 
 import redis
+import requests
 from fastapi import APIRouter, HTTPException
 
 from ubi_manifest import auth
 from ubi_manifest.worker.tasks.celery import app
 from ubi_manifest.worker.tasks.depsolve import depsolve_task
 
-from .models import DepsolveItem, DepsolverResult, DepsolverResultItem, TaskState
-from .utils import get_items_for_depsolving, get_repo_classes
+from .models import (
+    DepsolveItem,
+    DepsolverResult,
+    DepsolverResultItem,
+    StatusResult,
+    TaskState,
+)
+from .utils import get_gitlab_base_url, get_items_for_depsolving, get_repo_classes
+
+REQUEST_TIMEOUT = 20
 
 router = APIRouter(prefix="/api/v1")
 
 
-@router.get("/status")
-def status() -> dict[str, str]:
-    return {"status": "OK"}
+@router.get(
+    "/status",
+    response_model=StatusResult,
+    responses={
+        200: {
+            "content": {
+                "application/json": {
+                    "example": {
+                        "server_status": "OK",
+                        "workers_status": {
+                            "availability": {"worker01": {"ok": "pong"}},
+                            "stats": {"worker01": {"some": "stats"}},
+                            "registered_tasks": {"worker01": ["some_task"]},
+                            "active_tasks": {"worker01": []},
+                            "scheduled_tasks": {"worker01": []},
+                        },
+                        "redis_status": {"Connected": True},
+                        "celery_beat_status": "OK",
```

Comment:
the structure of  `celery_beat_status` is bit complicated now, can you please update it here?